examples/purchase-tester/src/main/java/com/revenuecat/purchasetester/OfferingFragment.kt

```diff
@@ -62,32 +66,77 @@ class OfferingFragment : Fragment(), PackageCardAdapter.PackageCardAdapterListen
     override fun onPurchasePackageClicked(cardView: View, currentPackage: Package, purchaseOption: PurchaseOption?) {
         binding.purchaseProgress.visibility = View.VISIBLE
 
-        if (purchaseOption == null) {
-            Purchases.sharedInstance.purchasePackageWith(
-                requireActivity(),
-                currentPackage,
-                { error, userCancelled ->
-                    if (!userCancelled) {
-                        showUserError(requireActivity(), error)
+        // TODO is this true? is it possible to have multiple subs to one app at once,
+        //  in which case we don't do an update but simply purchase again?
+        val isUpgrade = currentPackage.product.type == ProductType.SUBS && activeSubscriptions.isNotEmpty()
+
+        if (isUpgrade) {
+            // TODO how do we know which one to upgrade from if theres <1?
```

Comment:
the currentSubId is something developers pass in, so i guess that's on them

Reply:
It seems like the `currentSubId` might be something that needs to be used to calculate `isUpgraded` if they user belongs to multiple subscriptions? 🤔